Add modified time to webkitbugspy.

bls1999 opened the pull request:
#### 7a79cecafe05309d0d3308c479fd62be0827809c
<pre>
Add modified time to webkitbugspy.
<a href="https://bugs.webkit.org/show_bug.cgi?id=271272">https://bugs.webkit.org/show_bug.cgi?id=271272</a>
<a href="https://rdar.apple.com/125039020">rdar://125039020</a>

Reviewed by Jonathan Bedard.

Currently, webkitbugspy does not have a method to retrieve the last modified time of an Issue
object. This change adds this information into the Issue object such that it is callable.

Added getters/setters for the new modified property:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/issue.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/radar.py

Added the new modified property into the mocks:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/data.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/github.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/radar.py

Added new webkitbugspy unit tests:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/bugzilla_unittest.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/github_unittest.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/radar_unittest.py

Canonical link: <a href="https://commits.webkit.org/276428@main">https://commits.webkit.org/276428@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6117788364e9ee8cc5020fb57b15ee741c5ccfb4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44548 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23625 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46996 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47199 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40550 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46853 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27647 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/21015 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36634 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45124 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20682 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/38339 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17705 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/44420 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/18118 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39484 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2595 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40743 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39760 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48824 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19507 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/16040 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/43570 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20847 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/42322 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9940 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/21175 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20505 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/data.py:
```diff
@@ -49,6 +49,7 @@
     dict(
         title='Example issue 1',
         timestamp=1639536160,
+        modified=1710884407,
```
This is totally fine, although I might suggest making this number a little closer to the timestamp of the bug.

bls1999 replied:
Maybe silly, but for posterity, I always tend to use the current timestamp in my code when writing mock data. Kind of like a little easter egg for myself of, "oh hey, that's when I wrote that line!"

bls1999 committed:
Add modified time to webkitbugspy.
https://bugs.webkit.org/show_bug.cgi?id=271272
rdar://125039020

Reviewed by Jonathan Bedard.

Currently, webkitbugspy does not have a method to retrieve the last modified time of an Issue
object. This change adds this information into the Issue object such that it is callable.

Added getters/setters for the new modified property:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/issue.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/radar.py

Added the new modified property into the mocks:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/data.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/github.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/radar.py

Added new webkitbugspy unit tests:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/bugzilla_unittest.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/github_unittest.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/radar_unittest.py

Canonical link: https://commits.webkit.org/276428@main
